optimum/neuron/modeling_decoder.py

```diff
@@ -208,10 +208,15 @@ def _export(
             model_info = api.repo_info(model_id, revision=revision)
             checkpoint_revision = model_info.sha
 
+        if batch_size is None:
+            batch_size = 1
         # If the sequence_length was not specified, deduce it from the model configuration
         if sequence_length is None:
             # Note: for older models, max_position_embeddings is an alias for n_positions
             sequence_length = config.max_position_embeddings
+        if num_cores is None:
+            # Use all available cores
+            num_cores = len(os.listdir("/sys/class/neuron_device/")) * 2
```

Comment:
This will fail if the directory does not exist. Are we sure that this code can only be executed on a machine that has neuron devices?

Reply:
I added a check on top of the function. It is not possible to instantiate the `transformers-neuronx` model otherwise.